flocx_market/db/sqlalchemy/models.py

```diff
@@ -65,3 +66,48 @@ class Offer(Base):
         nullable=False,
     )
     cost = orm.Column(orm.Float, nullable=False)
+    contract_id = orm.Column(orm.String(64),
+                             orm.ForeignKey('contracts.contract_id'),
+                             nullable=True)
+    contract = orm.relationship('Contract')
+
+
+class Contract(Base):
+    __tablename__ = 'contracts'
+    contract_id = orm.Column(
+        orm.String(64),
+        primary_key=True,
+        autoincrement=False,
+    )
+    time_created = orm.Column(orm.DateTime(timezone=True), nullable=False)
+    status = orm.Column(orm.String(15), nullable=False, default='available')
+    start_time = orm.Column(orm.DateTime(timezone=True), nullable=False)
+    end_time = orm.Column(orm.DateTime(timezone=True), nullable=False)
+    cost = orm.Column(orm.Float, nullable=False)
+    bid_id = orm.Column(orm.String(64),
+                        orm.ForeignKey('bids.marketplace_bid_id'))
+    bid = orm.relationship('Bid')
+    offers = orm.relationship('Offer', lazy='dynamic')
+
+    def contract_to_dict(self):
+        d = {}
+        for col in self.__table__.columns:
+            val = getattr(self, col.name)
+            if type(val) == datetime.datetime:
+                d[col.name] = val.isoformat()
+            else:
+                d[col.name] = val
+
+        offer_model_objects = [offer for offer in self.offers.all()]
```

Comment:
This seems pretty dangerous; you're iterating through every single offer in the database, and if you're fetching, say, all contracts, you're doing that for every single contract.

I don't think this expansion needs to be done at all; just print the offer_id. I'd suggest removing the `contract_to_dict` function, and just using `to_dict`.

Reply:
Hey @tzumainn, yes, sure I can make those separate commits
I call self.offers.all(), that is only iterating through all the offers related to this contract, not all in the database. Should I expand the basic to_dict method or how do you think I should approach returning of offer_ids related to this contract?

Reply:
Whoops, you're right!

I don't think you should expand the offer at all here; the current to_dict method is sufficient for simply having the value for offer_id, correct?

Reply:
I'm not storing the offers in the contracts because there can be many offers that match 1 bid and we don't know the number. So instead I just keep a foreign key on offers for contracts and that's why I need to pull them with self.offers. What do you think it's the best solution in that case @tzumainn?

Reply:
Ah - I think there may be a misunderstanding here. We shouldn't need to keep track of all the offers that match a single bid; the matcher will assign one of those offers to that bid, and create a contract.

It's possible that we haven't gotten the data model quite right yet, so if we change it in the future, that's fine. Either way, I think we just want to use a generic `to_dict` function to return to users through the REST API. Don't worry about following relationships or anything yet.

Reply:
hmm okay, I'm not keeping track of them, I am receiving a list of offers from the matcher when a contract is being made, and then I update those offers to reference the contract that is passed in. But I can't keep the offers in the contract table since there can be many offers to one contract. In the API documentation it says that the contracts should return the list of offers that it is associated with, should I just leave that out then @tzumainn?

Reply:
I may have misunderstood; I thought you meant offers matching bids, but you mean offers matching contracts. But I thought that if a bid is looking for 5 nodes, then 5 separate contracts are created for each node - in other words, each contract specifies exactly one node?

Reply:
yes, sorry I was unclear there. Hmm, not sure, I was following the API documentation, there the contracts can have multiple offers and 1 bid only @tzumainn

Reply:
Hm, interesting, maybe I'm wrong! In any case, I think my general point stands: the to_dict method here should just change the database object into a dict; it shouldn't do any additional object expansion.